java/arcs/core/policy/Policy.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.policy
+
+import arcs.core.data.Annotation
+
+/** Defines a data usage policy. See [PolicyProto] for the canonical definition of a policy. */
+data class Policy(
+    val name: String,
+    val description: String,
+    val egressType: EgressType,
+    val targets: List<PolicyTarget>,
+    val configs: Map<String, PolicyConfig>,
+    val annotations: List<Annotation>
+)
+
+/** Target schema governed by a policy, see [PolicyTargetProto]. */
+data class PolicyTarget(
+    // TODO(b/157605232): Resolve the schema name to a type.
+    val schemaName: String,
+    val maxAgeMs: Long,
+    val retentions: List<PolicyRetention>,
+    val fields: List<PolicyField>,
+    val annotations: List<Annotation>
+)
+
+/** Allowed usages for fields in a schema, see [PolicyFieldProto]. */
+data class PolicyField(
+    // TODO(b/157605232): Resolve the field name to a type.
+    val fieldName: String,
+    /** Valid usages of this field without redaction. */
+    val rawUsages: Set<UsageType>,
```

Comment:
I think we decided to keep this separate in the proto. Do we need to do it in Kotlin too? Can we merge `redactedUsages` and `rawUsages` into a single `Map<String, Set<UsageType>>` with `"raw"` (or any other appropirate name) as the key for raw usage?

(Also, I don't remember why we decided to split off raw usage as a separate field. Can you remind me what it is?)

Reply:
The original reason was to separate a purely syntactic choice (that is, the choice to represent usage without redaction via the label "raw") from semantics. 

I wanted to pull them apart here based on how I anticipate they will be used, e.g.:

```
rawUsages.forEach { usage -> 
  addClaim("allowedFor$usage")
}
redactedUsages.forEach { (label, usage) -> 
  addClaim("allowedFor$usage_$label")
}
```

Reply:
Sounds good!